owncloudApp/src/main/java/com/owncloud/android/presentation/logging/LogsListActivity.kt

```diff
@@ -108,4 +120,46 @@ class LogsListActivity : AppCompatActivity() {
 
         recyclerViewLogsAdapter.setData(items)
     }
+
+    private fun uploadFile(file: File) {
+        val destinationPath = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
+        val destinationFolder = File(destinationPath)
+        val originalName = file.name
+        var uniqueName = originalName
+        var fileNumber = 1
+
+        while (File(destinationFolder, uniqueName).exists()) {
+            val fileNameParts = originalName.split(".")
+            uniqueName = "${fileNameParts[0]}_${fileNumber}.${fileNameParts[1]}.log"
+            fileNumber++
+        }
+
+        val destination = File(destinationFolder, uniqueName)
+
+        try {
+            val sourceChannel = FileInputStream(file).channel
+            val destinationChannel = FileOutputStream(destination).channel
+            sourceChannel.transferTo(0, sourceChannel.size(), destinationChannel)
+            sourceChannel.close()
+            destinationChannel.close()
+            showDownloadDialog(destination.name)
+
+        } catch (e: IOException) {
+            e.printStackTrace()
+            Timber.e(e, "There was a problem to download the file to Download folder.")
+        }
+    }
+
+    private fun showDownloadDialog(fileName: String) {
+        val dialog = AlertDialog.Builder(this)
+            .setTitle(getString(R.string.log_file_downloaded))
+            .setIcon(R.drawable.ic_baseline_download_grey)
+            .setMessage(getString(R.string.log_file_downloaded_description, fileName))
+            .setPositiveButton(android.R.string.ok) { dialog, _ ->
+                dialog.dismiss()
+                startActivity(Intent(DownloadManager.ACTION_VIEW_DOWNLOADS))
+            }
```

Comment:
We shouldn't have this action in the "OK" button (which could be named "CLOSE" as well), we should have an additional button with this action, for example "GO TO DOWNLOADS" or something like that

Reply:
Done